### PR TITLE
Update carbon-apimgt to 4.1.0 alpha version and Update integration test apim image to 4.1.0 alpha 

### PR DESCRIPTION
**Pull request by suksw**
####  Choreo Connect related
- Update carbon-apimgt version to align with product-apim 4.1.0 alpha
- Update APIM related certs
- Minor change to KeyManagerDiscoveryClient log

#### Choreo Connect tests related
- Update APIM image to 4.1.0-alpha and related certs
- Update test utils from APIM (`org.wso2.am.integration.common.test.utils`) to 4.1.0-alpha. 
- Update deployment.toml and docker-compose files used in tests for APIM. Update mounted DB files used for tests.
- Update CC test util methods. Use internal clients of APIM's RestAPIPublisherImpl and RestAPIStoreImpl to avoid the following methods for synapse gateway being triggered.
    - waitForDeployAPI()
    - waitForUnDeployAPI()
    - waitUntilSubscriptionAvailableInGateway()
    - waitUntilApplicationAvailableInGateway()
    - waitUntilApplicationKeyMappingAvailableInGateway()
    - waitUntilSubscriptionRemoveFromGateway()
- Remove currently unused test methods

### Issues
Fixes https://github.com/wso2/product-microgateway/issues/2715
Fixes https://github.com/wso2/product-microgateway/issues/2716

### Related PRs
https://github.com/wso2/product-microgateway/pull/2683

### Automation tests
 - Unit tests added: No
 - Integration tests added: Yes

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -484,7 +484,7 @@
 
         <andes.client.version>3.3.12</andes.client.version>
         <azure.messaging.eventhubs.version>5.6.0</azure.messaging.eventhubs.version>
-        <carbon.apimgt.version>9.18.12</carbon.apimgt.version>
+        <carbon.apimgt.version>9.18.2</carbon.apimgt.version>
```
Comment (slahirucd7) [inline pom.xml:487]: Is it fine to downgrade carbon-apimgt version here ? We updated the above version with M4.

**Review by praminda — APPROVED** [review]
Please check lahiru's comment. Rest of the changes looks ok to me

**Review by slahirucd7 — APPROVED** [review]

File: integration/test-integration/src/test/resources/dockerCompose/apim-in-common-network-docker-compose.yaml [hunk 2]
```diff
@@ -4,16 +4,16 @@ networks:
...
     ports:
       - "9763:9763"
       - "9443:9443"
       - "5672:5672"
       - "8243:8243"
     volumes:
-      - ./conf/deployment.toml:/home/wso2carbon/wso2am-4.0.0/repository/conf/deployment.toml
-      - ./database:/home/wso2carbon/wso2am-4.0.0/repository/database:rw
+      - ./conf/deployment.toml:/home/wso2carbon/wso2am-4.1.0-alpha/repository/conf/deployment.toml
+      - ./database:/home/wso2carbon/wso2am-4.1.0-alpha/repository/database:rw
```
Comment (renuka-fernando) [inline integration/test-integration/src/test/resources/dockerCompose/apim-in-common-network-docker-compose.yaml:16]: is this change required in the file [resources/docker-compose/apim/docker-compose.yaml](https://github.com/wso2/product-microgateway/pull/2714/files#diff-aa496dca1149563958a1b496c03c6258cda6a3536eb9e83e41b5880061e749a2) too, i think i forgot

Comment (renuka-fernando) [inline integration/test-integration/src/test/resources/dockerCompose/apim-in-common-network-docker-compose.yaml:16]:
```suggestion
      - ./conf/deployment.toml:/home/wso2carbon/wso2-config-volume/repository/conf/deployment.toml
      - ./database:/home/wso2carbon/wso2-artifact-volume/repository/database:rw
```
